### ISSUE TITLE
undefined method ts_msec

### ISSUE THREAD

**Issue by thushw** [dzunk/redis-time-series#74]
Hey thanks so much for this gem!

I noticed that `ts_msec` has been defined for `Time` but not for `ActiveSupport::TimeWithZone` and that is why I got this error:

```
irb(main):002:0> test_ts = Redis::TimeSeries.new('test', redis: $redis_enterprise)
=> #<Redis::TimeSeries:0x000000010c1c7b88 @key="test", @redis=#<Redis client v4.5.1 for redis://127.0.0.1:6379/0>>
irb(main):003:0> test_ts.add(1234, Time.current)
/Users/thushara/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/rails-fc4d83f43cd3/activesupport/lib/active_support/time_with_zone.rb:535:in `method_missing': undefined method `ts_msec' for Thu, 21 Jul 2022 17:41:05.380072000 UTC +00:00:Time (NoMethodError)

      wrap_with_time_zone time.__send__(sym, *args, &block)
                              ^^^^^^^^^
/Users/thushara/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bundler/gems/rails-fc4d83f43cd3/activesupport/lib/active_support/time_with_zone.rb:535:in `method_missing': undefined method `ts_msec' for 2022-07-21 17:41:05.380072 UTC:Time (NoMethodError)

      wrap_with_time_zone time.__send__(sym, *args, &block)
                              ^^^^^^^^^
irb(main):004:0>
```

Wondering if there is an easy fix/work-around?

**Comment by dzunk** [comment]
Hey, I took a quick look and wasn't able to reproduce on the latest version of the gem. Here's what my environment looks like right now:
```ruby
[16] pry(main)> RUBY_VERSION
=> "3.0.2"
[17] pry(main)> Redis::VERSION
=> "4.7.1"
[18] pry(main)> Redis::TimeSeries::VERSION
=> "0.7.0"
[19] pry(main)> ActiveSupport.version
=> Gem::Version.new("6.1.6.1")
[20] pry(main)> @ts1
=> #<Redis::TimeSeries:0x00007f8afc23f148 @key="ts1", @redis=#<Redis client v4.7.1 for redis://127.0.0.1:6379/0>>
[21] pry(main)> @ts1.add(1234, Time.current)
=> #<Redis::TimeSeries::Sample:0x00007f8afd9e8768 @time=2022-07-21 14:00:50 449839/1048576 -0500, @value=0.1234e4>
```
Can you share your versions as well?

**Comment by dzunk** [comment]
Never mind, I just reproduced - it's because the return value of `Time.current` changes depending on whether or not `Time.zone` is set (https://github.com/rails/rails/blob/6-0-stable/activesupport/lib/active_support/core_ext/time/calculations.rb#L39)
```ruby
[38] pry(main)> Time.zone = nil
=> nil
[39] pry(main)> @ts1.add(1234, Time.current)
=> #<Redis::TimeSeries::Sample:0x00007f8afda03978 @time=2022-07-21 14:05:22 187695/1048576 -0500, @value=0.1234e4>
[40] pry(main)> Time.zone = 'UTC'
=> "UTC"
[41] pry(main)> @ts1.add(1234, Time.current)
NoMethodError: undefined method `ts_msec' for Thu, 21 Jul 2022 19:05:26.571629000 UTC +00:00:Time
from /Users/dzunk/.gem/ruby/3.0.2/gems/activesupport-6.1.6.1/lib/active_support/time_with_zone.rb:535:in `method_missing'
Caused by NoMethodError: undefined method `ts_msec' for 2022-07-21 19:05:26.571629 UTC:Time
from /Users/dzunk/.gem/ruby/3.0.2/gems/activesupport-6.1.6.1/lib/active_support/time_with_zone.rb:535:in `method_missing'
```
As a workaround you can call `.to_time` on the `TimeWithZone` object:
```ruby
[50] pry(main)> @ts1.add(1234, Time.current)
NoMethodError: undefined method `ts_msec' for Thu, 21 Jul 2022 19:07:46.762628000 UTC +00:00:Time
from /Users/dzunk/.gem/ruby/3.0.2/gems/activesupport-6.1.6.1/lib/active_support/time_with_zone.rb:535:in `method_missing'
Caused by NoMethodError: undefined method `ts_msec' for 2022-07-21 19:07:46.762628 UTC:Time
from /Users/dzunk/.gem/ruby/3.0.2/gems/activesupport-6.1.6.1/lib/active_support/time_with_zone.rb:535:in `method_missing'
[51] pry(main)> @ts1.add(1234, Time.current.to_time)
=> #<Redis::TimeSeries::Sample:0x00007f8afda48910 @time=2022-07-21 14:07:48 2080375/2097152 -0500, @value=0.1234e4>
```
That should just work though - will get that fixed up!

**Comment by dzunk** [comment]
Released this as 0.7.1 - try out that version and see if it's fixed 😄
